### PR TITLE
Replace pkg_resources with importlib.metadata

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

Address #1672 

If I understood it correctly this PR should not be merged before we drop Python 3.7 suppoort next week on Dec. 26, 2021.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
> If I understood it correctly this PR should not be merged before we drop Python 3.7 suppoort next week on Dec. 26, 2021.

Yes, do you want to open two PRs dropping supports of old Python and NumPy versions.

- Drop NumPy 1.18 on Dec. 22, 2022 (xref https://github.com/GenericMappingTools/pygmt/pull/1430)
- Drop Python 3.7 on Dec. 26, 2022 (xref https://github.com/GenericMappingTools/pygmt/pull/699)

File: pygmt/__init__.py [hunk 1]
```diff
@@ -19,8 +19,7 @@
 """
 
 import atexit as _atexit
-
-from pkg_resources import get_distribution
+from importlib.metadata import version
```
Comment (weiji14) [inline pygmt/__init__.py:22]: So `pylint` is complaining about the `version` variable being redefined:

```
pylint pygmt setup.py doc/conf.py
************* Module pygmt
pygmt/__init__.py:137:16: W0621: Redefining name 'version' from outer scope (line 22) (redefined-outer-name)
pygmt/__init__.py:150:12: W0621: Redefining name 'version' from outer scope (line 22) (redefined-outer-name)
```

Maybe rename `version` to `gs_version` and `gmt_version` at L137/L140 and L150/L153 respectively.

File: pygmt/__init__.py [hunk 2]
```diff
@@ -63,7 +62,7 @@
 )
 
 # Get semantic version through setuptools-scm
-__version__ = f'v{get_distribution("pygmt").version}'  # e.g. v0.1.2.dev3+g0ab3cd78
+__version__ = version("pygmt")  # e.g. v0.1.2.dev3+g0ab3cd78
```
Comment (seisman) [inline pygmt/__init__.py:65]: The `version` function returns the version string without the leading `v` (e.g., `0.1.2`):
```suggestion
__version__ = f'v{version("pygmt")}'  # e.g. v0.1.2.dev3+g0ab3cd78
```

File: pygmt/__init__.py [hunk 3]
```diff
@@ -135,10 +134,10 @@ def _get_ghostscript_version():
 
         for gs_cmd in cmds:
             try:
-                version = subprocess.check_output(
+                gs_version = subprocess.check_output(
```
Comment (seisman) [inline pygmt/__init__.py:137]:
```suggestion
                return subprocess.check_output(
```

File: pygmt/__init__.py [hunk 4]
```diff
@@ -135,10 +134,10 @@ def _get_ghostscript_version():
...
                     [gs_cmd, "--version"], universal_newlines=True
                 ).strip()
-                return version
+                return gs_version
```
Comment (seisman) [inline pygmt/__init__.py:140]:
```suggestion
```

File: pygmt/__init__.py [hunk 5]
```diff
@@ -148,10 +147,10 @@ def _get_gmt_version():
         Get GMT version.
         """
         try:
-            version = subprocess.check_output(
+            gmt_version = subprocess.check_output(
```
Comment (seisman) [inline pygmt/__init__.py:150]:
```suggestion
            return subprocess.check_output(
```

File: pygmt/__init__.py [hunk 6]
```diff
@@ -148,10 +147,10 @@ def _get_gmt_version():
...
                 ["gmt", "--version"], universal_newlines=True
             ).strip()
-            return version
+            return gmt_version
```
Comment (seisman) [inline pygmt/__init__.py:153]:
```suggestion
```

**Review by seisman — APPROVED** [review]

**Review by weiji14 — APPROVED** [review]
:rocket:

**Pull request merged**